go-controller/pkg/ovn/admin_network_policy_test.go

```diff
@@ -1099,6 +1100,111 @@ var _ = ginkgo.Describe("OVN ANP Operations", func() {
+						Subresource: "status",
+					},
+					{
+						Manager: "node2",
+						Operation: metav1.ManagedFieldsOperationApply,
+						APIVersion: "policy.networking.k8s.io/v1alpha1",
+						FieldsType: "FieldsV1",
+						FieldsV1: &metav1.FieldsV1{Raw: []byte(`{"f:status":{"f:conditions":{"k:{\"type\":\"Ready-In-Zone-node2\"}":{".":{},"f:lastTransitionTime":{},"f:message":{},"f:reason":{},"f:status":{},"f:type":{}}}}}`)},
+						Subresource: "status",
+					},
+				}
+				anp, err := fakeOVN.fakeClient.ANPClient.PolicyV1alpha1().AdminNetworkPolicies().Create(context.TODO(), anp, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Eventually(func() int {
+					anp, err = fakeOVN.fakeClient.ANPClient.PolicyV1alpha1().AdminNetworkPolicies().Get(context.TODO(), "harry-potter", metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(anp.Status.Conditions)
+				}).Should(gomega.Equal(1))
+				klog.Infof("SURYA %v", anp.GetManagedFields())
+				gomega.Expect(anp.Status.Conditions[0].Type).To(gomega.Equal("Ready-In-Zone-"+node1Name))
+				gomega.Expect(anp.Status.Conditions[0].Message).To(gomega.Equal("Setting up OVN DB plumbing was successful"))
+				gomega.Expect(anp.Status.Conditions[0].Reason).To(gomega.Equal("SetupSucceeded"))
+				gomega.Expect(anp.Status.Conditions[0].Status).To(gomega.Equal(metav1.ConditionTrue))
+
+				// Simulate zone (node2) adding a status to this ANP using server side apply
+				// This is because in UT we cannot simulate multi-zones
+				ginkgo.By("2. Update ANP status to also include status from node2 (simulation)")
+				readyCondition := metav1.Condition{
+					Type:    "Ready-In-Zone-node2",
+					Status:  metav1.ConditionTrue,
+					Reason:  "SetupSucceeded",
+					Message: "Setting up OVN DB plumbing was successful",
+				}
+				applyObj := anpapiapply.AdminNetworkPolicy(anp.Name).
+					WithStatus(anpapiapply.AdminNetworkPolicyStatus().WithConditions(readyCondition))
+				_, err = fakeOVN.fakeClient.ANPClient.PolicyV1alpha1().AdminNetworkPolicies().
+					ApplyStatus(context.TODO(), applyObj, metav1.ApplyOptions{FieldManager: node2Name, Force: true})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				klog.Infof("SURYA %v", anp)
+				gomega.Eventually(func() int {
+					anp, err = fakeOVN.fakeClient.ANPClient.PolicyV1alpha1().AdminNetworkPolicies().Get(context.TODO(), "harry-potter", metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(anp.Status.Conditions)
+				}).Should(gomega.Equal(2))
+				klog.Infof("SURYA %v", anp)
+				gomega.Expect(anp.Status.Conditions[0].Type).To(gomega.BeElementOf("Ready-In-Zone-"+node1Name, "Ready-In-Zone-"+node2Name))
+				gomega.Expect(anp.Status.Conditions[1].Type).To(gomega.BeElementOf("Ready-In-Zone-"+node1Name, "Ready-In-Zone-"+node2Name))
+
+				/*ginkgo.By("3. Delete node1 and ensure the status get's updated correctly")
```

Comment:
let the records show that I tried to add UT and miserably failed because fakeClients are not working well with apply status, after spending 4 hours trying to read various upstream things I have decided to just add e2e's.
PS: It is so sad that no one else has written a node add/ node delete upstream and that it is template-less haha which is why I thought UT would be easier, but ofc nature doesn't want me to do anything easily ;)